ams/ams-server/src/main/java/com/netease/arctic/ams/server/service/impl/OptimizerService.java

```diff
@@ -226,5 +236,52 @@ private Optimizer fillContainerType(Optimizer optimizer) {
     optimizer.setContainerType(container.getType());
     return optimizer;
   }
+
+  /**
+   * If lastmodification is not null and is not the same as before, then the task has occurred retry and returns true.
+   */
+  private void checkOptimizerRetry(OptimizerStateReport newReportData,Optimizer oldOptimizer) {
+
+    Map<String, String> stateInfo = oldOptimizer.getStateInfo();
+    if (stateInfo == null || newReportData.optimizerState == null ||
+            newReportData.optimizerState.get(STATUS_IDENTIFICATION) == null) {
+      return;
+    }
+
+    //If there is no status identifier in optimizerState, it is skipped
+    String statusIdentification = newReportData.optimizerState
+            .get(STATUS_IDENTIFICATION);
+    if (!statusIdentification.equals(stateInfo.get(STATUS_IDENTIFICATION)) &&
+            stateInfo.get(STATUS_IDENTIFICATION) != null) {
+
+      LOG.info("Optimizer " + newReportData.optimizerId + " retry occurred");
+      //The flink status has changed. Procedure
+      long optimizerId = newReportData.optimizerId;
+      try (SqlSession sqlSession = getSqlSession(true)) {
+        OptimizeTasksMapper optimizeTasksMapper = getMapper(sqlSession, OptimizeTasksMapper.class);
+        //Gets the task in execution of the optimizer
+        List<BaseOptimizeTask> baseOptimizeTasks = optimizeTasksMapper
+                .selectOptimizeTasksByJobIDAndStatus(optimizerId, OptimizeStatus.Executing.name());
+        for (BaseOptimizeTask baseOptimizeTask : baseOptimizeTasks) {
+
+          //Since optimzer may poll tasks before heartbeat,
+          //only tasks whose createtime is less than status Identification are set to fail
+          if (baseOptimizeTask.getCreateTime()  < Long.parseLong(statusIdentification)) {
```

Comment:
Checking `create time of task` < `last modify time of optimizer` is not safe, I recommend checking `execute time of task` < `last modify time of optimizer`.

We can get the execute time of task by 
```
OptimizeTaskItem optimizeTask = XXX;
optimizeTask.getOptimizeRuntime().getExecuteTime();
```